src/pyggdrasil/tree_inference/_file_id.py

```diff
@@ -0,0 +1,206 @@
+"""Provides classes for naming files Tree,
+Cell Simulation and MCMC run files uniquely """
+
+from enum import Enum
+from typing import Union, Optional
+
+
+from pyggdrasil.tree_inference import CellAttachmentStrategy, McmcConfig
+
+
+class TreeType(Enum):
+    """Enum representing valid tree types implemented in pyggdrasil.
+
+    Allowed values:
+      - RANDOM (random tree)
+      - STAR (star tree)
+      - DEEP (deep tree)
+      - HUNTRESS (Huntress tree) - inferred from real / cell simulation data
+    """
+
+    RANDOM = "r"
+    STAR = "s"
+    DEEP = "d"
+    HUNTRESS = "h"
+
+
+class MutationDataId:
+    """Class representing a mutation data id.
+
+    In case we want to infer a tree from real data,
+    we need to provide a mutation data id.
+    """
+
+    id = str
+
+
+class TreeId:
+    """Class representing a tree id.
+
+    A tree id is a unique identifier for a tree.
+
+    tree_type: TreeType - type of tree
+    n_nodes: int - number of nodes in the tree
+    seed: int - seed used to generate the tree
+    cell_simulation_id: str - if the tree was generated from a cell
+                                simulation, i.e. Huntress
+    """
+
+    tree_type: TreeType
+    n_nodes: int
+    seed: Optional[int]
+    cell_simulation_id: Union[MutationDataId, None]
+
+    id: str
+
+    def __init__(
+        self,
+        tree_type: TreeType,
+        n_nodes: int,
+        seed: Optional[int] = None,
+        cell_simulation_id: Optional[MutationDataId] = None,
+    ):
+        """Initializes a tree id.
+
+        Args:
+            tree_type: TreeType
+            n_nodes: int
+            seed: int
+        """
+
+        self.tree_type = tree_type
+        self.n_nodes = n_nodes
+        self.seed = seed
+        self.cell_simulation_id = cell_simulation_id
+
+        self.id = self._create_id()
+
+    def _create_id(self) -> str:
```

Comment:
It may stay this way, but it'd be cleaner not to have side effects in this method (as above you have `self.id = self._create_id()`, it could only return a value).

Reply:
Yes, I see -  changed it.